lms/services/application_instance.py

```diff
@@ -1,34 +1,41 @@
-from typing import Optional
+from functools import lru_cache
 
 from lms.models import ApplicationInstance
+from lms.services import ConsumerKeyError
 
 
 class ApplicationInstanceService:
-    def __init__(self, db):
+    def __init__(self, db, default_consumer_key):
         self._db = db
+        self._default_consumer_key = default_consumer_key
 
-    def get(self, consumer_key: str) -> Optional[ApplicationInstance]:
-        """Query one ApplicationInstance by consumer_key."""
+    @lru_cache
+    def get(self, consumer_key=None) -> ApplicationInstance:
+        """
+        Return the ApplicationInstance with the given consumer_key.
 
-        return ApplicationInstance.get_by_consumer_key(self._db, consumer_key)
+        If no consumer_key is given return the current request's
+        ApplicationInstance (the ApplicationInstance whose consumer_key matches
+        request.lti_user.oauth_consumer_key).
 
-    @staticmethod
-    def update_settings(ai, canvas_sections_enabled=None, canvas_groups_enabled=None):
+        :raise ConsumerKeyError: if the consumer_key isn't in the database
         """
-        Update ApplicationInstance.settings.
+        consumer_key = consumer_key or self._default_consumer_key
 
-        For not nullable values only changes the params passed with a not None value
-        """
-        if canvas_sections_enabled is not None:
-            ai.settings.set("canvas", "sections_enabled", canvas_sections_enabled)
+        application_instance = ApplicationInstance.get_by_consumer_key(
+            self._db, consumer_key
+        )
```

Comment:
`get()` now raises `ConsumerKeyError` if the `consumer_key` isn't found. That's what `ApplicationInstanceGetter` did, and the user code expects it. Also `consumer_key` is optional and defaults to the one from the request (very useful)